knative-operator/pkg/controller/knativeservingobsolete/knativeservingobsolete_controller.go

```diff
@@ -120,9 +203,14 @@ func (r *ReconcileKnativeServingObsolete) Reconcile(request reconcile.Request) (
 func (r *ReconcileKnativeServingObsolete) removeOldCertsConfig(ns string) error {
 	const name = "controller"
 	deployment := &appsv1.Deployment{}
-	if err := r.client.Get(context.TODO(), client.ObjectKey{Namespace: ns, Name: name}, deployment); err != nil {
+	err := r.client.Get(context.TODO(), client.ObjectKey{Namespace: ns, Name: name}, deployment)
+	if errors.IsNotFound(err) {
```

Comment:
This is important for non-upgrade cases, where the users (because they have old scripts or whatnot) still apply the old CR without having a system around. We can safely ignore this condition anyway.